python/pyspark/ml/wrapper.py

```diff
@@ -71,6 +71,9 @@ def _new_java_array(pylist, java_class):
         """
         Create a Java array of given java_class type. Useful for
         calling a method with a Scala Array from Python with Py4J.
+        If the param pylist is a 2D array, then a 2D java array will be returned.
+        The returned 2D java array is a square, non-jagged 2D array that is big
+        enough for all elements.
```

Comment:
The slots in the inner Java arrays will be filled with null to make the non-jagged 2D array. I feel that it is better to mention it.